processor/spanmetricsprocessor/processor.go

```diff
@@ -379,7 +447,27 @@ func buildKey(serviceName string, span pdata.Span, optionalDims []Dimension) met
 		if attr, ok := spanAttr.Get(d.Name); ok {
 			value = tracetranslator.AttributeValueToString(attr)
 		}
-		concatDimensionValue(&metricKeyBuilder, value, true)
+		concatDimensionValue(&metricKeyBuilder, value)
+	}
+
+	k := metricKey(metricKeyBuilder.String())
+	return k
+}
+
+func buildResourceAttrKey(serviceName string, optionalResourceAttrs []Dimension, resourceAttr pdata.AttributeMap) metricKey {
+	var metricKeyBuilder strings.Builder
+	concatDimensionValue(&metricKeyBuilder, serviceName)
+
+	var value string
+	for _, ra := range optionalResourceAttrs {
+		// Set the default if configured, otherwise this metric will have no value set for the resource attribute.
+		if ra.Default != nil {
+			value = *ra.Default
+		}
+		if attr, ok := resourceAttr.Get(value); ok {
+			value = tracetranslator.AttributeValueToString(attr)
+		}
+		concatDimensionValue(&metricKeyBuilder, value)
```

Comment:
we can't make sure the sequence will be the same every time. We need to order them before we concatenate them.

Reply:
Because we are reading off `optionalResourceAttrs` which is stored statically in the processor structure as `resourceAttributes` it should always maintain the same order. This isn't programmed defensively tho, it's just implied.

The original function assumes the same thing above, I think sorting will add time complexity when we don't need to. Maybe we can just add a comment to state this as a warning? Thoughts?

Reply:
How about we change this function to be a method so that `p.resourceAttributes` will be always used inside this method?

Reply:
Yeah will do, this sgtm